tests/typescript.test.mjs

```diff
@@ -0,0 +1,187 @@
+    let result = await execa('pnpm', ['lint'], {
+      cwd: join(tmpDir.path, appName),
+    });
+
+    console.log(result.stdout);
+  });
+
+  it('glint passes', async function () {
+    let result = await execa('pnpm', ['glint'], {
+      cwd: join(tmpDir.path, appName),
+    });
+
+    console.log(result.stdout);
+  });
+
+  it('successfully runs tests', async function () {
+    let result;
+
+    try {
+      result = await execa('pnpm', ['test:ember', '--test-port', '0'], {
+        cwd: join(tmpDir.path, appName),
+      });
+    } catch (err) {
+      console.log(err.stdout, err.stderr);
+      throw err;
+    }
+
+    // make sure that each of the tests that we expect actually show up
+    // alternatively we can change this to search for `pass 3`
+    expect(result.stdout).to.contain(
+      'Acceptance | welcome page: visiting /index shows the welcome page',
+    );
+    expect(result.stdout).to.contain('Acceptance | styles: visiting /styles');
+
+    console.log(result.stdout);
+  });
+
+  it('successfully runs tests in dev mode', async function () {
+    await execa({
+      cwd: join(tmpDir.path, appName),
+    })`pnpm install --save-dev testem http-proxy`;
+    let appURL;
+
+    let server;
+
+    try {
+      server = execa('pnpm', ['start'], {
+        cwd: join(tmpDir.path, appName),
+      });
+
+      await new Promise((resolve) => {
+        server.stdout.on('data', (line) => {
+          let result = /Local:\s+(https?:\/\/.*)\//g.exec(
+            stripAnsi(line.toString()),
+          );
+
+          if (result) {
+            appURL = result[1];
+            resolve();
+          }
+        });
+      });
+
+      writeFileSync(
+        join(tmpDir.path, appName, 'testem-dev.js'),
+        `module.exports = {
+  test_page: 'tests/index.html?hidepassed',
+  disable_watching: true,
+  launch_in_ci: ['Chrome'],
+  launch_in_dev: ['Chrome'],
+  browser_start_timeout: 120,
+  browser_args: {
+    Chrome: {
+      ci: [
+        // --no-sandbox is needed when running Chrome inside a container
+        process.env.CI ? '--no-sandbox' : null,
+        '--headless',
+        '--disable-dev-shm-usage',
+        '--disable-software-rasterizer',
+        '--mute-audio',
+        '--remote-debugging-port=0',
+        '--window-size=1440,900',
+      ].filter(Boolean),
+    },
+  },
+  middleware: [
+    require(__dirname + '/testem-proxy.js')('${appURL}')
+  ],
+};
+`,
+      );
+
+      let testResult = await execa(
+        'pnpm',
+        ['testem', '--file', 'testem-dev.js', 'ci', '--port', '0'],
+        {
+          cwd: join(tmpDir.path, appName),
+        },
+      );
+      expect(testResult.exitCode).to.eq(0, testResult.output);
+    } finally {
+      server?.kill('SIGINT');
+    }
+  });
+
+  it('successfully optimizes deps', function () {
+    return execa('pnpm', ['vite', 'optimize', '--force'], {
+      cwd: join(tmpDir.path, appName),
+    });
+  });
+});
```

Comment:
I feel like this file is 90% the same as the other one 🤔 I wonder if we can reuse the tests themselves but only change the setup between JS and TS?

Reply:
yeah, I'll see what happens -- aside from the fixture filess needing to be different (the plan is to convert them all to TS for the ts tests), we're basically just checking that the package.json scripts succeed, and that's shareable.